src/core/source_buffers/source_buffers_store.ts

```diff
@@ -162,6 +144,34 @@ export default class SourceBuffersStore {
     this._onNativeSourceBufferAddedOrDisabled = [];
   }
 
+  /**
+   * Get all currently available buffer types.
+   * /!\ This list can evolve at runtime depending on feature switching.
+   * @returns {Array.<string>}
+   */
+  public getBufferTypes() : IBufferType[] {
+    const bufferTypes : IBufferType[] = this.getNativeBufferTypes();
+    if (features.nativeTextTracksBuffer != null ||
+        features.htmlTextTracksBuffer != null
+    ) {
+      bufferTypes.push("text");
+    }
+    if (features.imageBuffer != null) {
+      bufferTypes.push("image");
+    }
+    return bufferTypes;
+  }
+
+  /**
+   * Get all "native" buffer types that should be created before beginning to
+   * push contents.
+   * @returns {Array.<string>}
+   */
+  public getNativeBufferTypes() : IBufferType[] {
```

Comment:
The nodeName check could be done once in constructor, no ?

Reply:
Yeah but I found it more readable that way and I don't think this is performance-sensitive